src/pages/resources/custom-containers.mdx

```diff
@@ -0,0 +1,182 @@
+  final container.
+</Note>
+
+<CodeGroup>
+
+```docker {{ title: "Python" }}
+FROM python:3.11-slim
+
+ARG HANDLER
+
+ENV HANDLER=${HANDLER}
+ENV PYTHONUNBUFFERED=TRUE
+
+RUN apt-get update -y && \
+    apt-get install -y ca-certificates && \
+    update-ca-certificates
+
+RUN pip install --upgrade pip pipenv
+
+# Copy either requirements.txt or Pipfile
+COPY requirements.tx[t] Pipfil[e] Pipfile.loc[k] ./
+
+# Guarantee lock file if we have a Pipfile and no Pipfile.lock
+RUN (stat Pipfile && pipenv lock) || echo "No Pipfile found"
+
+# Output a requirements.txt file for final module install if there is a Pipfile.lock found
+RUN (stat Pipfile.lock && pipenv requirements > requirements.txt) || echo "No Pipfile.lock found"
+
+RUN pip install --no-cache-dir -r requirements.txt
+
+COPY . .
+
+ENTRYPOINT python $HANDLER
+```
+
+```docker {{ title: "JavaScript" }}
+# syntax=docker/dockerfile:1
+FROM node:alpine
+
+ARG HANDLER
+ENV HANDLER=${HANDLER}
+
+RUN apk update && \
+    apk add --no-cache ca-certificates && \
+    update-ca-certificates
+
+# Python and make are required by certain native package build processes in NPM packages.
+ENV PYTHONUNBUFFERED=1
+RUN apk add --update --no-cache python3 make g++ && ln -sf python3 /usr/bin/python
+RUN python3 -m ensurepip
+RUN pip3 install --no-cache --upgrade pip setuptools
+
+COPY . .
+
+RUN \
+  set -ex; \
+  yarn install --production --frozen-lockfile --cache-folder /tmp/.cache; \
+  rm -rf /tmp/.cache; \
+  # prisma fix for docker installs: https://github.com/prisma/docs/issues/4365
+  test -d ./prisma && npx prisma generate || echo "";
+
+ENTRYPOINT node $HANDLER
+```
+
+```docker {{ title: "TypeScript" }}
+# syntax=docker/dockerfile:1
+FROM node:alpine as build
+
+ARG HANDLER
+
+# Python and make are required by certain native package build processes in NPM packages.
+RUN apk add g++ make py3-pip
+RUN yarn global add typescript @vercel/ncc
+
+WORKDIR /usr/app
+
+COPY package.json yarn.lock ./
+
+RUN yarn install --production --frozen-lockfile --cache-folder /tmp/.cache && \
+    rm -rf /tmp/.cache
+
+COPY . .
+
+RUN \
+  --mount=type=cache,target=/tmp/ncc-cache \
+  ncc build ${HANDLER} -o lib/ -t
+
+FROM node:alpine as final
+
+WORKDIR /usr/app
+
+RUN apk update && \
+    apk add --no-cache ca-certificates && \
+    update-ca-certificates
+
+COPY package.json yarn.lock ./
+
+RUN yarn install --production --frozen-lockfile --cache-folder /tmp/.cache && \
+    rm -rf /tmp/.cache
+
+COPY . .
+
+COPY --from=build /usr/app/lib/ ./lib/
+
+ENTRYPOINT ["node", "lib/index.js"]
+```
+
+```docker {{ title: "Go" }}
+FROM golang:alpine as build
+
+ARG HANDLER
+
+WORKDIR /app/
+
+COPY go.mod *.sum ./
+
+RUN go mod download
+
+COPY . .
+
+RUN go build -o /bin/main ./${HANDLER}/...
+
+FROM alpine
+
+COPY --from=build /bin/main /bin/main
+
+RUN chmod +x-rw /bin/main
+RUN apk update && \
+    apk add --no-cache tzdata ca-certificates && \
+    update-ca-certificates
+
+ENTRYPOINT ["/bin/main"]
+```
+
```

Comment:
```suggestion
```

Reply:
remove golang for now I think (until we have SDK docs for it

Reply:
I added it to provide an example of a compiled language, we could swap for C# or something if you think that's better for now

Reply:
nah all good, we can leave it in on that basis